src/gui/mainwindow.h

```diff
@@ -21,6 +24,9 @@ class MainWindow : public QMainWindow {
     ~MainWindow();
 
 private:
+    std::unique_ptr<QWidget> MainWidget;
```

Comment:
You probably took a example from below, but usage of unique_ptr is wrong here. Qt widget and windows took ownership of the pointer so we always should pass raw pointers for such cases. Anyway this is just a side note as these fields will disappear once you will rebase with master :)